Added support for eth, added mapping from rsk to eth.

valgaro opened the pull request:
Added initial files for eth support and a mapping in the index file to
use etc when the coin type is 'rsk'.
Ticket: BG-32

Signed-off-by: Gonzalo Alvarez <alvarez14@avaya.com>

Commit:
Added support for eth, added mapping from rsk to eth.

Added initial files for etc support and a mapping in the index file to
use etc when the coin type is 'rsk'.
Ticket: BG-32

Signed-off-by: Gonzalo Alvarez <alvarez14@avaya.com>

Commit:
Added skipped tests for ETH, added new behavior for transaction building

Ticket: BG-32

Signed-off-by: Gonzalo Alvarez <alvarez14@avaya.com>

Commit:
Merge changes

Commit:
Support isValidAddress

Commit:
Added initial build implementation for transaction

Added the implementation for build and the transaction constructor to
create an Eth transaction.

Ticket: BG-36

Signed-off-by: Gonzalo Alvarez <alvarez14@avaya.com>

silalvado committed:
ETH utils unit tests update

silalvado committed:
Merge branch 'BG-32-eth-rsk-initial' of github.com:paszyrko/bitgo-account-lib into BG-32-eth-rsk-initial

Commit:
Added integration with signature and improvements

Added integration between utils.js and signature.js, added logic to
avoid error in canSign, added new methods for parameters in hex format.

Ticket: BG-37

Signed-off-by: Gonzalo Alvarez <alvarez14@avaya.com>

silalvado committed:
Add function getAddress to ETH keypair and fix ETH utils unit test

Commit:
Added integration with getAddress and fixed tests

Added integration to get the address from the private key, added library
dependency to convert toHex.

Ticket: BG-38

Signed-off-by: Gonzalo Alvarez <alvarez14@avaya.com>

silalvado committed:
Modification of getAddress function to obtain address from public key or private key

silalvado committed:
Simplification de ETH getAddress and update unit test for ETH utils

Commit:
Improved tests using getAddress, added walletUtil for constants

Ticket: BG-39

Signed-off-by: Gonzalo Alvarez <alvarez14@avaya.com>

paszyrko commented on package.json:
```diff
@@ -42,7 +42,12 @@
     "protobufjs": "^6.8.9",
-    "tronweb": "^2.7.2"
+    "tronweb": "^2.7.2",
+    "web3": "^1.2.6",
```
Add a comment saying that this will be removed later

valgaro replied:
I can't comment the json but I will do it in utils.js

paszyrko commented on src/coin/eth/enum.ts:
```diff
@@ -0,0 +1,17 @@
+  /**
+   * This is the multi-sig contract configuration.
+   */
+  WalletConfiguration,
```
Remove this entry

valgaro replied:
Done

paszyrko commented on src/coin/eth/signature.ts:
```diff
@@ -0,0 +1,316 @@
+    fields.forEach((field, i) => {
+      self._fields.push(field.name);
+      /**
+       *
```
clean these comments

paszyrko commented on src/coin/eth/utils.ts:
```diff
@@ -0,0 +1,315 @@
+const web3 = new Web3(rskDomain);
+
+/**
+ * Signs the transaction using the Eth elliptic curve
```
General comment. I think it is necessary to add documentation to the methods :(

Commit:
Fixed PR comments

Removed unused transaction type, added comment to remove dependency later.

Ticket: BG-37
Signed-off-by: Gonzalo Alvarez <gonzaalvarez70@gmail.com>

valgaro committed:
Removed unused comments in signature.ts

Signed-off-by: Gonzalo Alvarez <gonzaalvarez70@gmail.com>
